### PR TITLE
Support deleting resources on unreachable cluster

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #2033

Introduces `kubernetes:deleteUnresponsive` and `PULUMI_K8S_DELETE_UNREACHABLE`
as ways of being able to specify that you want to delete resources
when a cluster is unreachable

[asciicast](https://asciinema.org/a/0Xwo1OaCs4YWfxL07V4F3xZck)

### CODE REVIEW

File: provider/pkg/gen/schema.go [hunk 1]
```diff
@@ -68,6 +68,10 @@ func PulumiSchema(swagger map[string]interface{}) pschema.PackageSpec {
 					Description: "If present, the default namespace to use. This flag is ignored for cluster-scoped resources.\n\nA namespace can be specified in multiple places, and the precedence is as follows:\n1. `.metadata.namespace` set on the resource.\n2. This `namespace` parameter.\n3. `namespace` set for the active context in the kubeconfig.",
 					TypeSpec:    pschema.TypeSpec{Type: "string"},
 				},
+				"deleteUnreachable": {
+					Description: "If present and set to true, the provider will delete resources on an unreachable kubernetes cluster",
```
Comment (lblackstone) [inline provider/pkg/gen/schema.go:72]: I'd suggest rewording this slightly since we aren't actually deleting the resources on the cluster.
```suggestion
					Description: "If present and set to true, the provider will delete resources associated with an unreachable kubernetes cluster from Pulumi state",
```

**Review by lblackstone — APPROVED** [review]
LGTM aside from the message wording 🎉

File: provider/pkg/gen/schema.go (hunk 1 above)
Comment (ericrudder) [reply provider/pkg/gen/schema.go:72]:
```suggestion
					Description: "If present and set to true, the provider will delete resources on an unreachable Kubernetes cluster",
```
